### PR TITLE
Refactor hooks, use esbuild

### PR DESCRIPTION
**Pull request by popeindustries**
- swap Rollup for esbuild
- add `onDependencyBundle` hook to override cjs bundling
- add default `onTransform` hook behaviour to support TypeScript/JSX without a custom transformer
- remove deprecated `--transpiler` option
- bundle `dvlp` with esbuild, and reduce production dependencies to 2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,26 +34,22 @@ $ npm install dvlp
...
-  -s, --silent                suppress default logging
-  --no-reload                 disable reloading connected browsers on file change
-  -t, --transpiler <path>     [deprecated] path to optional transpiler file
-  -v, --version               output the version number
-  -h, --help                  display help for command
+  -p, --port <port>   port number
+  -m, --mock <path>   path to mock files (directory, file, glob pattern)
+  -k, --hooks <path>  path to optional hooks registration file
+  --ssl <path>        enable https mode by specifying path to directory containing
+                      ".crt" and ".key" files
```
Comment (simenbrekken) [inline README.md:48]: Should we add back "(directory or glob pattern)"?

File: README.md [hunk 2]
```diff
@@ -74,7 +70,7 @@ $ npm run dev
 
 ### Hooks
 
-In some cases, you may want to write source code in a non-standard, higher-order language like SASS (for CSS) or JSX (for JS), or modify a response body before sending to the browser. In these cases, you can register `hooks` to convert file contents on the fly when imported by an application server or requested by the browser.
+In some cases, source code may need to be transformed into a valid format before it is executed, or a response body modified before sending to the browser. In these cases, you can register `hooks` to convert file contents on the fly when imported by an application server or requested by the browser.
```
Comment (simenbrekken) [inline README.md:73]: I think this should be "sending it to the"

**Review by simenbrekken — APPROVED** [review]
Didn't get a change chance to run this yet but I'll try and link it early next week and give it a whirl

File: README.md (hunk 1 above)
Comment (popeindustries) [reply README.md:48]: Ah, sorry. This was generated by cli `dvlp --help`. Will add to both...

**Pull request merged**